src/routes/sign/Sign.svelte

```diff
@@ -1,7 +1,43 @@
 <script lang="ts">
     import Transaction from '../../lib/transaction/Transaction.svelte';
+    import { getParamsFromUrl } from './signHelpers';
+    import { xdr, description, isWaiting } from './signStore';
+
+    const queryString = window.location.search;
+    const urlParams = getParamsFromUrl(queryString);
+
+    if (urlParams) {
+        $xdr = urlParams.xdr;
+        $description = urlParams.description;
+    }
+
+    setTimeout(() => {
+        if (!$xdr) {
+            $isWaiting = false;
+        }
+    }, 2000);
+
+    window.addEventListener('message', (e) => {
+        if (e.origin !== 'https://localhost:3000') {
```

Comment:
What's this supposed to be? Simple Stellar Signer should not contain references to any particular origin.